compiler/damlc/daml-stdlib-src/DA/Upgrade.daml

```diff
@@ -88,6 +88,14 @@ instance (MetaEquiv c1 c2, GenConvertible f1 f2) => GenConvertible (M1 i1 c1 f1)
 instance (GenConvertible a1 a2, GenConvertible b1 b2) => GenConvertible (a1 :*: b1) (a2 :*: b2) where
   cv ~(P1 a b) = P1 (cv a) (cv b)
 
+-- product embeddings
+instance (GenConvertible a1 a2) => GenConvertible a1 (a2 :*: Opt b s) where
```

Comment:
I totally agree with this instance. It doesn't necessarily work when you add the new field _not_ as the last field of the record, right? I suppose that is intended.

Reply:
yes, that's intended.